src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/common/GetFormatDataCommand.cs

```diff
@@ -97,9 +97,14 @@ protected override void BeginProcessing()
         /// </summary>
         protected override void ProcessRecord()
         {
-            bool writeOldWay = PowerShellVersion == null ||
-                               PowerShellVersion.Major < 5 ||
-                               (PowerShellVersion.Major == 5 && PowerShellVersion.Minor < 1);
+            // During remoting, remain compatible with older clients by default.
+            // Remoting detection: Automatic variable $PSSenderInfo is defined in true remoting contexts and also in background jobs.
+            //                     However, its Boolean .PSShowComputerName NoteProperty is true only in remoting contexts.
+            bool inRemoteSession = (GetVariableValue("PSSenderInfo") as PSObject)?.Properties["PSShowComputerName"]?.Value is bool value && value;
```

Comment:
This doesn't look right to me.  The `PSSenderInfo` variable in the session is always of `PSSenderInfo` type.  It is only a `PSObject` type if it goes through the serialization system.  So retrieving the built in variable here as `PSObject` type will always be null and `inRemoteSession` will always be false.

The only way I can think of distinguishing between remote session and background job (remote) session is to change the `PSSenderInfo.ConnectionString` to something different for background jobs (since it is meaningless for background jobs), maybe just `PSSenderInfo.ConnectionString = "BackgroundJob"`.

Reply:
That sounds like a fairly effective solution. 🤔

Reply:
We have a sample:
https://github.com/PowerShell/PowerShell/blob/ee4eba5348739b6ef17cf2069685cc2d1815a62e/src/System.Management.Automation/engine/GetCommandCommand.cs#L492-L504

Reply:
You're right, @PaulHigin. Apart from the `PSObject` problem, looking for this property on the _server_ side is the wrong approach anyway.

Setting `.ConnectionString` to something different makes perfect sense - I would just set it to the empty string, given that there's no connection involved.

Currently, it it is `https://localhost` - any concerns about changing it?

Reply:
@iSazonov: The linked approach doesn't distinguish between true remoting and background jobs, which we need to make.

Reply:
@mklement0 I still believe the check should be made on the server side.  If I understand the problem correctly, we need to use the 'old format way' when talking remotely to older clients.  So I think we can make the right decision on the server by looking at the PSSenderInfo.  Something like:

```
bool writeOldWay = false;
var psSenderInfo = GetVariableValue("PSSenderInfo");
if (psSenderInfo != null)
{
    var clientPSVersion = psSenderInfo.ApplicationArguments?.PSVersionTable?.PSVersion
    writeOldWay = (clientPSVersion != null) && (clientVersion.Major < 5 || (clientVersion.Major == 5 && clientVersion.Minor < 1));
}
...
```

Reply:
@PaulHigin: Absolutely, the test needs to be on the server side (the problem with my previous attempt was to approach the problem from the client side).

Thanks for pointing me to the fact that `$psSenderInfo.ApplicationArguments` actually contains the client `$PSVersionTable`.

I've rewritten the test as follows:

* If `-PowerShellVersion` is used to explicitly indicate a client version, use it compare it to `5.1` - irrespective of whether the invocation is local or not.

* Otherwise, if `$PSSenderInfo` is present, examine `$psSenderInfo.ApplicationArguments.PSVersionTable.PSVersion` - i.e., the remoting client's version - and test that against `5.1`. Treat the case where `$PSSenderInfo` is present, but the client version cannot be extracted as the pre-`5.1` case, to be safe (not sure if that could ever happen).

* Otherwise (purely local invocation without `-PowerShellVersion`), return _all_ formats.

It's fine to then not to have to distinguish between true remoting and background jobs.

The above leads me  to think that introducing `-PowerShellVersion` may never have been needed, because the client version information could always have been derived from `$psSenderInfo.ApplicationArguments.PSVersionTable.PSVersion` (or its absence), correct?

---

Separately - and this will have to become separate issues:

I still think it would make sense to have `.ConnectionString` reflect whether code is executing in a remote session or in a background job.

* You can _almost_ infer it already, namely by testing for `http://localhost`, whereas WinRM-based invocations contain a port number and client version number; e.g., `http://localhost:5985/wsman?PSVersion=5.1.18362.145

* However, `http://localhost` is _also_ the value in _SSH_-based remoting sessions - which is a problem in itself. If this gets fixed, and there's a concern that `.ConnectionString` should otherwise not be changed for the sake of backward compatibility, `http://localhost` could be used (though it would be unexpected, given that there's no HTTP transport involved, I presume).

Reply:
Please see #11292 and #11293.